qiskit/compiler/transpile.py

```diff
@@ -201,10 +218,21 @@ def callback_func(**kwargs):
         config = user_config.get_config()
         optimization_level = config.get('transpile_optimization_level', 1)
 
+    if scheduling_method is None:
+        if any_delay_in(circuits):
+            raise TranspilerError("Invalid arguments: When transpiling circuits with delays, "
+                                  "'scheduling_method' is required.")
+    else:
+        if backend is None:
+            if not basis_gates or not instruction_durations:
```

Comment:
I think this is a little more to the point: if we have a scheduling method specified, but backend is none and we don't have the required info to continue with the scheduling pass
```suggestion
        if backend is None and not (basis_gates and instruction_durations):
```